src/Windows/Markdown/Editor/Impl/ContainedLanguage/RExpressionTermFilter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information
+
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.Diagnostics;
+using Microsoft.Languages.Core.Text;
+using Microsoft.Languages.Editor.Document;
+using Microsoft.Languages.Editor.Text;
+using Microsoft.R.Core.Parser;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.Markdown.Editor.ContainedLanguage {
+    /// <summary>
+    /// Helps R parser to handle R Markdown-specific constructs
+    /// </summary>
+    internal sealed class RExpressionTermFilter: IExpressionTermFilter {
+        private readonly ITextBuffer _textBuffer;
+
+        public RExpressionTermFilter(ITextBuffer textBuffer) {
+            Check.ArgumentNull(nameof(textBuffer), textBuffer);
+            _textBuffer = textBuffer; // R language buffer
+            _textBuffer.AddService(this);
```

Comment:
It doesn't need to be removed. Generally removing services is optional unless they are disposable or complicated things that can introduce leaks such as when attached to VS COM events.